meta/Page.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace dokuwiki\plugin\struct\meta;
+
+class Page {
+
+    /** @var \helper_plugin_sqlite */
+    protected $sqlite;
+
+    protected $pid;
+    protected $title = null;
+    protected $lasteditor = null;
+
+    public function __construct($pid) {
+        /** @var \helper_plugin_struct_db $helper */
+        $helper = plugin_load('helper', 'struct_db');
+        $this->sqlite = $helper->getDB();
+        $this->pid = $pid;
+    }
+
+    /**
+     * Load the title from the database
+     *
+     * @return null|string the current title, null if not exists
+     */
+    protected function loadTitle() {
+        $sql = "SELECT title FROM titles WHERE pid = ?";
+        $res = $this->sqlite->query($sql, $this->pid);
+        $title = $this->sqlite->res2single($res);
+        $this->sqlite->res_close($res);
+
+        if($title === false) return null;
+        return $title;
+    }
+
+    protected function loadLastEditor() {
+        $sql = "SELECT lasteditor FROM titles WHERE pid = ?";
+        $res = $this->sqlite->query($sql, $this->pid);
+        $lasteditor = $this->sqlite->res2single($res);
+        $this->sqlite->res_close($res);
+
+        if($lasteditor === false) return null;
+        return $lasteditor;
+    }
+
+    protected function loadLastRevision() {
+        $sql = "SELECT lastrev FROM titles WHERE pid = ?";
+        $res = $this->sqlite->query($sql, $this->pid);
+        $lastrev = $this->sqlite->res2single($res);
+        $this->sqlite->res_close($res);
+
+        if($lastrev === false) return null;
+        return $lastrev;
+    }
```

Comment:
I think it would be more effective if this class would always load all three values from the database. This way when I request a title first and then the editor, it would make only one database request.